overlord/snapstate/snapstate.go

```diff
@@ -1118,7 +1123,12 @@ func Update(st *state.State, name, channel string, revision snap.Revision, userI
 	}
 
 	params := func(update *snap.Info) (string, Flags, *SnapState) {
-		return channel, flags, &snapst
+		updateFlags := flags
+		if !update.NeedsClassic() && updateFlags.Classic {
```

Comment:
Slightly sad that our code requires to duplicate this with the code above.

Reply:
Actually given that infoForUpdate is also changed (below) - do we need this change here or is the change in infoForUpdate sufficient?

Reply:
Snap info is verified twice, once in `infoForUpdate()` and then once again in `doUpdate()`. I'm beginning to think that maybe verifying flags in `infoForUpdate()` is actually superfluous.

Reply:
@bboozzoo Thanks for checking this! Would be great if you could double check, the less code the better :)

Reply:
Dropped the part in `infoForUpdate()`.